.github/workflows/automerge.yml

```diff
@@ -23,9 +22,30 @@ jobs:
           PR_URL: ${{github.event.pull_request.html_url}}
           GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
 
-      - name: Enable auto-merge for Dependabot PRs
+      - name: Enable Auto-Merge for Dependabot PRs
         if: steps.metadata.outputs.update-type == 'version-update:semver-minor' || steps.metadata.outputs.update-type == 'version-update:semver-patch'
         run: gh pr merge --auto --merge "$PR_URL"
         env:
           PR_URL: ${{github.event.pull_request.html_url}}
           GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+  automerge-pre-commit:
+    runs-on: ubuntu-latest
+    if: github.actor == 'pre-commit-ci[bot]'
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Enable Auto-Approve for Pre-Commit
+        if: github.head_ref == 'pre-commit-ci-update-config'
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{github.event.pull_request.html_url}}
+          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Enable Auto-Merge for Pre-Commit
+        if: github.head_ref == 'pre-commit-ci-update-config'
+        run: gh pr merge --auto --merge "$PR_URL"
```

Comment:
Nice! only problem is that we require projects to have automerge enabled but is fine